File: backend/reservation-service/src/main/java/com/quisin/reservation/client/RestaurantServiceClient.java
package com.quisin.reservation.client;

import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;

@FeignClient(name = "restaurant-service", path = "/api/v1/restaurants")
public interface RestaurantServiceClient {

    @GetMapping("/{restaurantId}/tables")
    List<TableDto> getRestaurantTables(@PathVariable String restaurantId);

    @GetMapping("/{restaurantId}/tables/{tableId}")
    TableDto getTable(
        @PathVariable String restaurantId,
        @PathVariable String tableId
    );

    @GetMapping("/{restaurantId}/tables/available")
    List<TableDto> getAvailableTables(
        @PathVariable String restaurantId,
        @RequestParam int capacity
    );

    @GetMapping("/{restaurantId}/operating-hours/check")
    boolean isRestaurantOperating(
        @PathVariable String restaurantId,
        @RequestParam String dateTime
    );
} 

File: backend/menu-service/src/main/kotlin/com/quisin/menu/client/RestaurantServiceClient.kt
package com.quisin.menu.client

import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import java.time.LocalDateTime

@FeignClient(name = "restaurant-service", url = "\${quisin.services.restaurant-service.url}")
interface RestaurantServiceClient {

    @GetMapping("/api/v1/restaurants/{restaurantId}")
    fun getRestaurant(@PathVariable restaurantId: String): RestaurantResponse
}

data class RestaurantResponse(
    val id: String,
    val name: String,
    val description: String,
    val status: RestaurantStatus,
    val ownerId: String,
    val chainId: String?,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)

enum class RestaurantStatus {
    PENDING,
    ACTIVE,
    INACTIVE,
    SUSPENDED
} 